version.swift

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/swift
+
+import Foundation
+
+/// Tag management
+
+struct Tag: CustomStringConvertible {
+    let major: Int
+    let minor: Int
+    let patch: Int
+
+    var description: String {
+        return "\(major).\(minor).\(patch)"
+    }
+
+    func nextPatch() -> Tag {
+        return Tag(major: major, minor: minor, patch: patch + 1)
+    }
+
+    func nextMinor() -> Tag {
+        return Tag(major: major, minor: minor + 1, patch: 0)
+    }
+
+    func nextMajor() -> Tag {
+        return Tag(major: major + 1, minor: 0, patch: 0)
+    }
+}
+
+final class TagManager {
+    func loadCurrentTag() -> Tag? {
+        let regex = "^[0-9]*\\.[0-9]*\\.[0-9]"
```

Comment:
I think you would also need a complete stop, aka. `$`, at the end of this regex. Just in case someone names their tag `4.49.1-rc1`.

Reply:
Could potentially also specify that you need at least 1 digit in each segment.

I.e.
```swift
let regex = "^\\d+\\.\\d+\\.\\d+$"
```

Reply:
Grep's syntax doesn't support notations like \d. But I changed it to
```swift
let regex = "^[0-9]\\+\\.[0-9]\\+\\.[0-9]\\+$"
```

Reply:
Ah, right. For that to be allowed you need to add the `-E` argument, aka. extended regex.

But the one you changed to is more than sufficient 👌